rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,154 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+* **Hc**: each key is hashed to a checksum which is used to validate the inserted key during the deconstruction phase.
+* **data types**: the data types for *count*, *val_sum* and *hash_sum* MUST match ((un)signed, byte-size).
+
+The following parameters are used:
+
+* **buckets**: 1024
+* **Hc**: [murmur3](https://en.wikipedia.org/wiki/MurmurHash), 8 bytes, seed: 0x00
+* **Hk**: murmur3, 4 bytes, seed: 0x01, k = 6 
+* **key sum size**: 32 bytes
+* **bucket count field**: 4 bytes
+
+See Appendix A.1 for an explanation.
+
+### 4.1.3 IBLT page size
+
+Calculating an IBLT is relatively cheap, but it grows linearly with the number of transactions. This would harm performance of larger networks.
+The immutable nature of transactions make it possible to store intermediate IBLTs and use those as an optimization.
+The LC range parameters of the protocol should be adjusted accordingly to optimize use of these intermediate IBLTs.
+Therefore, they should not be chosen freely.
+
+Any LC value used in the protocol MUST be a multiple of 512 (minus 1, because the LC is 0 based). 
+The use of inclusive or exclusive range parameters must make sure that `end-start` is a multiple of 512.
+In the context of an IBLT this size will be referred to as the `IBLT page size` or `PSIZE` for short.
+
+### 4.2 Operation
+
+All messages and values mentioned in this chapter are scoped to a single connection between two nodes.
+*Alice* and *Bob* are used to represent two nodes connected to each other.
+
+The protocol generally operates as follows:
+
+1. Alice broadcasts at a set interval (§4.2.1):
+    * XOR of all known transaction references.
+    * Highest Lamport Clock value over all transactions (LC).
+    
+2. When receiving Alice's broadcast, Bob compares the XOR value from Alice with its own (§4.2.2):
+    * When the XOR is the same and the LC value equals BOB's highest Lamport Clock value: no action required.
+    * When different, send a message containing:
+      * LC value sent by Alice
+      * the IBLT that includes the Lamport Clock range of 0-LC
+      * Bob's highest Lamport Clock value
+
+3. When receiving the message, Alice subtracts Bob's IBLT from her own IBLT for the given range (§4.2.3):
+   * If not deconstructable, go to 1 and send values based on a lower Lamport Clock value.
+   * If deconstructable, send a request for missing transactions
+   * If Bob's highest Lamport Clock value is higher that the LC value sent in 1, then request transactions by Lamport Clock value:
+     * Lamport Clock start value.
+     * Lamport Clock end value.
+
+4. When receiving a request for transactions, Bob responds with a message including the requested transactions (§4.2.4 and §4.2.5).
+
+5. After adding Bob's transactions to the DAG \(making sure its cryptographic signature is valid\), query the payload if it's missing.
+
+A node MUST make sure to only add transactions of which all previous transactions are present.
```

Comment:
Does the protocol above assure that transactions are sync'd in order?

Reply:
That requirement has been added to 4.2.5 Transaction List message